src/graphnet/models/task/task.py

```diff
@@ -22,14 +23,30 @@ class Task(LightningModule):
     def nb_inputs(self) -> int:
         """Number of inputs assumed by task."""
 
-    def __init__(self, hidden_size: int, target_label: str, loss_function: LossFunction):
+    def __init__(
+        self, 
+        hidden_size: int, 
+        target_label: str, 
+        loss_function: LossFunction,
+        transform_prediction_and_target: Optional[Callable] = None,
+        transform_target: Optional[Callable] = None,
+        transform_inference: Optional[Callable] = None,
+    ):
         # Base class constructor
         super().__init__()
 
+        # Check(s)
+        assert not((transform_prediction_and_target is not None) and (transform_target is not None)), \
+            "Please specify at most one of `transform_prediction_and_target` and `transform_target`"
+        assert (transform_target is not None) == (transform_inference is not None), \
+            "Please specify both `transform_inference` and `transform_target`"
+
```

Comment:
Maybe here it would be good to add a sanity check that `transform_target` and `transform_inference` are indeed inverse? Could potentially help catch a few bugs.
```suggestion
        if transform_target is not None:
            x_test = np.logspace(-6, 6, 12 + 1)
            x_test = torch.from_numpy(np.concatenate([-x_test[::-1], [0], x_test]))
            t_test = transform_target(x_test)
            valid = torch.isfinite(t_test)
            assert torch.allclose(transform_inference(t_test)[valid], x_test[valid]), \
                "The provided transforms for targets during training and predictions during inference at not inverse."
            del x_test, t_test, valid
```

Reply:
This is clever, I like it